### ISSUE TITLE
Open preCommands for bot admins

### ISSUE THREAD

**Issue by Chumper** [renovatebot/renovate#12147]
### What would you like Renovate to be able to do?

I would like to have the option to add custom commands to the Renovate sidecars before executing the actual command.

Let us assume the following config:
```js
{
  binarySource: 'docker',
  ...
}
```
and Renovate running on a Go repository with the following repo config:
```json
{
  "postUpdateOptions": [
    "gomodTidy"
  ],
  ...
}
```
Renovate will now go ahead, update the dependencies and update any lock files as well.
In this case it will also try to tidy the `go.mod` file and will add some preCommands to the container: https://github.com/renovatebot/renovate/blob/main/lib/manager/gomod/artifacts.ts#L29
This is needed because go may need to resolve package from GitHub.

In our setup we need to add the same for the GitHub Enterprise server.

### If you have any ideas on how this should be implemented, please tell us here.

Yes, 

I would like to open the `preCommands` to bot admins as well.
Either scoped to each manager (in this case the go manager) or globally.

That would enable bot admins like me to better cater the needs of repository with internal registries.

This would somehow cross concerns with the `customEnvVariables` because potentially you could also set env vars in the `preCommands` but that is something I would accept for now.

### Is this a feature you are interested in implementing yourself?

Yes

**Comment by rarkins** [comment]
So would it be something like a global/admin "allowPreUpdateCommands" option as a boolean or perhaps regex, and then allow users to configure `preUpdateCommands` as an array?

**Comment by Chumper** [comment]
Closed in favor of #11470
